p2p/nat/nat_stun.go

```diff
@@ -30,14 +30,18 @@ import (
 const STUNDefaultServerAddr = "159.223.0.83:3478"
 
 type STUN struct {
-	serverAddr string
+	serverAddr *net.UDPAddr
```

Comment:
perhaps rename `serverAddr` to `server`. It's already an `UDPAddr`, so having type in the name is a bit stuttery